File: app/src/main/java/com/cpma/app/repository/DeviceAPI.java
package com.cpma.app.repository;

import com.cpma.app.model.request.BatteryRequest;
import com.cpma.app.model.request.DeviceRequest;
import com.cpma.app.model.response.JWTResponse;
import com.cpma.app.model.request.LocationRequest;
import com.cpma.app.model.request.LoginRequest;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.PUT;

public interface DeviceAPI {
    @POST("api/auth/signin")
    Call<JWTResponse> login(@Body LoginRequest loginQuery);

    @PUT("api/android/saveDevice")
    Call<ResponseBody> saveDevice(@Body DeviceRequest deviceRequest);

    @POST("api/android/saveLocation")
    Call<ResponseBody> saveLocation(@Body LocationRequest locationRequest);

    @POST("api/android/saveBattery")
    Call<ResponseBody> saveBattery(@Body BatteryRequest batteryRequest);
}

File: app/src/main/java/com/cpma/app/repository/DeviceAPI.kt
package com.cpma.app.repository

import com.cpma.app.model.request.BatteryRequest
import com.cpma.app.model.request.DeviceRequest
import com.cpma.app.model.request.LocationRequest
import com.cpma.app.model.request.LoginRequest
import com.cpma.app.model.response.JWTResponse
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.PUT

interface DeviceAPI {
    companion object {
        const val BASE_URL = "http://192.168.0.150:8080/"
    }

    @POST("api/auth/signin")
    fun login(@Body loginQuery: LoginRequest): Call<JWTResponse>

    @PUT("api/android/saveDevice")
    fun saveDevice(@Body deviceRequest: DeviceRequest): Call<ResponseBody>

    @POST("api/android/saveLocation")
    fun saveLocation(@Body locationRequest: LocationRequest): Call<ResponseBody>

    @POST("api/android/saveBattery")
    fun saveBattery(@Body batteryRequest: BatteryRequest): Call<ResponseBody>
}